### DISCUSSION TITLE
Using Chem.SanitizeMol() appropriately

### DISCUSSION THREAD

**Discussion by tsnguyen05** [rdkit/rdkit#4406] [Q&A]
Greetings,

I am trying to follow using Chem.SanitizeMol from here: https://www.rdkit.org/docs/Cookbook.html (Error Messages section)., may I ask whether calling Chem.SanitizeMol() means that we turn off Sanitization? How can we turn it back on again after finishing using it? Furthermore, do we need to call mol.UpdatePropertyCache(strict=False) before calling Chem.SanitizeMol()? The reason is that I am trying to make a radical on the structure, "BrC(CC)CC(O)=O" (trying to create a radical at the ternary location) . Without using Chem.SantizeMol(), I received an error: 

RuntimeError: Pre-condition Violation
	getNumImplicitHs() called without preceding call to calcImplicitValence()
	Violation occurred on line 190 in file Code/GraphMol/Atom.cpp
	Failed Expression: d_implicitValence > -1
	RDKIT: 2021.03.4
	BOOST: 1_74

mol = "BrC(CC)CC(O)=O"
Thus, I decided to use the following lines:
Chem.SanitizeMol(mol)
mol.UpdatePropertyCache(strict=False)
mol = Chem.AddHs(mol)

On the next step, I start to use Chem.FragmentOnBonds(). It does not matter whether I set "AddDummies = True/False", I always receive "WARNING: not removing hydrogen atom with dummy atom neighbors." So that last question is that is there any way that I can resolve this issue? I looked at different old issues + discussions to resolve the radical problem, but none of them worked for me. I believe that I might have done something incorrectly! 

Thanks,
Tommy

**Comment by greglandrum** [comment]
Hi, in order to help us help you, please share the exact code (along with input data) that you executed and the output you saw from running that.

**Reply by tsnguyen05** [reply]
Hello Greg,

After several day searching this issue, I found a solution to fix it.
